### DISCUSSION TITLE
Map nested data before applying the schema in external data source

### DISCUSSION THREAD

**Discussion by gehtalles-de** [Budibase/budibase#1604] [Ideas]
Hi! My graphQL endpoint gives me a nested response:

```
{
  "data":  {
    "orders": [
      {
         "id": 1
       },
       {
         "id": 2
       }
    ]
  }
}

```
I'd like to be able to map this data with handlebars (like `{{ response.data.orders }}`) before I'm passing it to the schema tab.

**Comment by joebudi** [comment]
Thank you for your idea. I believe you spoke with @shogunpurple about this and it would be a nice feature.

**Comment by hillar** [comment]
same with elasticsearch `GET /my-index-000001/_search?q="{{q}}"`

```
{
  "took": 580,
  "timed_out": false,
  "_shards": {
    "total": 11,
    "successful": 11,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 10000,
      "relation": "gte"
    },
    "max_score": 17.113823,
    "hits": [
      {
        "_index": "my-index-000001",
        "_type": "_doc",
        "_id": "785f94fa1865320786a9fe30ab9740e1",
        "_score": 17.113823,
        "_source": { ...
```

**Reply by shogunpurple** [reply]
Hey @hillar  - we will be working on this over the coming weeks, allowing you to map data from external data sources.

You can follow along here:
https://github.com/Budibase/budibase/issues/2093
